src/state/slices/opportunitiesSlice/types.ts

```diff
@@ -43,9 +43,9 @@ export type UserStakingOpportunity = {
 }
 
 // The AccountId of the staking contract in the form of chainId:accountAddress
-export type StakingId = Nominal<string, 'StakingId'>
+export type StakingId = Nominal<string, 'StakingId'> & AssetId
 // The AccountId of the LP contract in the form of chainId:accountAddress
-export type LpId = Nominal<string, 'LpId'>
+export type LpId = Nominal<string, 'LpId'> & AssetId
```

Comment:
The guts of this PR, StakingId/LpId is an AssetId with an additional flavoring.
More flavor = tastes better = blazingly safe as long as you exercise